src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseGitDiffReportService.java

```diff
@@ -158,6 +160,32 @@ public ProgrammingExerciseGitDiffReport getOrCreateReportOfExercise(ProgrammingE
         }
     }
 
+    /**
+     * Calculates git diff between two repositories and returns the cumulative number of diff lines.
+     *
+     * @param urlRepoA       url of the first repo to compare
+     * @param localPathRepoA local path to the checked out instance of the first repo to compare
+     * @param urlRepoB       url of the second repo to compare
+     * @param localPathRepoB local path to the checked out instance of the second repo to compare
+     * @return cumulative number of lines in the git diff of given repositories
+     */
+    public int calculateNumberOfDiffLinesBetweenRepos(VcsRepositoryUrl urlRepoA, Path localPathRepoA, VcsRepositoryUrl urlRepoB, Path localPathRepoB) {
+        var repoA = gitService.getExistingCheckedOutRepositoryByLocalPath(localPathRepoA, urlRepoA);
+        var repoB = gitService.getExistingCheckedOutRepositoryByLocalPath(localPathRepoB, urlRepoB);
+
+        var treeParserRepoA = new FileTreeIterator(repoA);
+        var treeParserRepoB = new FileTreeIterator(repoB);
+
+        try (var diffOutputStream = new ByteArrayOutputStream(); var git = Git.wrap(repoB)) {
+            git.diff().setOldTree(treeParserRepoB).setNewTree(treeParserRepoA).setOutputStream(diffOutputStream).call();
+            var diff = diffOutputStream.toString();
+            return extractDiffEntries(diff).stream().mapToInt(ProgrammingExerciseGitDiffEntry::getLineCount).map(Math::abs).sum();
+        }
+        catch (IOException | GitAPIException e) {
+            return Integer.MAX_VALUE;
```

Comment:
Can you also log this error, so the admins can find potential future problems?

Reply:
Just got them from my translator @MarkusPaulsen :D

Reply:
Good idea